### PR TITLE
Pre-sort input types array in addTypesToUnion

### PR DESCRIPTION
**Pull request by jakebailey**

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot perf test this faster

**Comment by jakebailey** [comment]
Seems like no real change but let me try the more annoying pre-check.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -16251,6 +16251,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     // Add the given types to the given type set. Order is preserved, duplicates are removed,
     // and nested types of the given kind are flattened into the set.
     function addTypesToUnion(typeSet: Type[], includes: TypeFlags, types: readonly Type[]): TypeFlags {
+        // Pre-sort the input by ID so we deduplicate below.
+        types = types.slice().sort((a, b) => a.id - b.id);
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:16255]: You'd have to import it but:

```suggestion
        if (!arrayIsSorted(types) {
            types = types.slice().sort((a, b) => a.id - b.id);
        }
```

**Comment by jakebailey** [comment]
@typescript-bot perf test this faster

**Comment by jakebailey** [comment]
Meh, doesn't seem like it helps, at least not in our current benchmarks. Maybe it would help in the esoteric case I had, if I could find it, anyway...

**Pull request closed without merging**